yarn-project/noir-protocol-circuits/src/crates/types/src/abis/kernel_circuit_public_inputs.nr

```diff
@@ -28,15 +38,22 @@ struct KernelCircuitPublicInputsBuilder {
 impl KernelCircuitPublicInputsBuilder {
     pub fn finish(self) -> KernelCircuitPublicInputs {
         KernelCircuitPublicInputs {
+            aggregation_object: self.aggregation_object,
+            meta_hwm: self.meta_hwm,
             end: self.end.finish(),
             constants: self.constants,
             is_private: self.is_private,
         }
     }
 
     pub fn to_final(self) -> KernelCircuitPublicInputsFinal {
+        let meta = self.end_meta.to_final();
+        let app = self.end.to_final();
         KernelCircuitPublicInputsFinal {
-            end: self.end.to_final(),
+            aggregation_object: self.aggregation_object,
+            meta_hwm: self.meta_hwm,
+            end_meta: meta,
```

Comment:
For clarity, this is where the extra gates are coming from. As Alex mentioned, specializing the FinalAccumulatedData to meta and app will pare back the gates significantly, and could be incorporated in this PR.